MyFridge/MyFridge/Refrigerator.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyFridge
+{
+    internal class Refrigerator
+    {
+        private Shelf shelf;
+
+        public static int UniqueId { get; set; } = 1;
+        public int Id { get; } = 134;
+        public string Model { get; set; }
+        public string Color { get; set; }
+        public List<Shelf> Shelves { get; set; }
+        public int AmountOfShelves { get; set; }
+
+
+
+        public Refrigerator(string model, string color, List<Shelf> shelves)
+        {
+            Id += UniqueId++;
+            Model = model;
+            Color = color;
+            if (shelves == null) throw new Exception("shelves null");
+            foreach (var shelf in shelves)
+            {
+               if (shelf==null) throw new Exception("shelf null");
+            }
+            Shelves = shelves;
+            AmountOfShelves = Shelves.Count;
+        }
+
+
+
+        public override string ToString()
+        {
+            string s="";
+            foreach (var item in Shelves)
+            {
+                s += item.ToString();
+            }
+            return
+            ("id:" + Id + " model: " + Model + " color: " + Color + " amount of shelves: " + AmountOfShelves +
+             " shelves: " + s);
+        }
+
+        public int GetFreeSpace() { 
+            int freeSpace = 0;
+            foreach (var shelf in Shelves)
+            {
+                freeSpace += shelf.SpaceAvailable;
+            }
+            return freeSpace;
+        }
+
+        public void InsertingItem(Item item)
+        {
+            if (item == null) throw new Exception("invalid item");
+            if (GetFreeSpace()<item.Space) throw new Exception("There is not enough space");
+            foreach (var shelf in Shelves)
+            {
+                if (shelf.SpaceAvailable>=item.Space)
+                {
+                    shelf.InsertingItem(item);
+                    return;
+                }
+                throw new Exception("There is not enough space");
```

Comment:
This logic is incorrect. Try to think why :)

say item is 10 cm
say shelf1 has only 5 cm left
and shelf2 has 20 cm left.

please fix